submission/persistence-sql/src/main/kotlin/ac/uk/ebi/biostd/persistence/integration/services/SqlPersistenceService.kt

```diff
@@ -23,13 +24,17 @@ internal open class SqlPersistenceService(
     }
 
     @Transactional
-    override fun getSequenceNextValue(pattern: String): Long = lockExecutor.executeLocking(pattern) {
-        val sequence = sequenceRepository.findByPrefix(pattern) ?: throw SequenceNotFoundException(pattern)
-        var next = sequence.counter.count + 1
-        while (submissionPersistenceQueryService.existByAccNo("${sequence.prefix}$next")) next++
+    override fun getSequenceNextValue(pattern: String): Long {
+        suspend fun getNextSequence(pattern: String): Long {
+            val sequence = sequenceRepository.findByPrefix(pattern) ?: throw SequenceNotFoundException(pattern)
+            var next = sequence.counter.count + 1
+            while (submissionPersistenceQueryService.existByAccNo("${sequence.prefix}$next")) next++
 
-        sequence.counter.count = next
-        sequenceRepository.save(sequence).counter.count
+            sequence.counter.count = next
+            val saved = sequenceRepository.save(sequence)
+            return saved.counter.count
+        }
+        return lockExecutor.executeLocking(pattern) { runBlocking { return@runBlocking getNextSequence(pattern) } }
```

Comment:
I think it looks clearer leaving it as it's right now and just adding the `runBlocking`. Also, once we migrate to mysql reactive, we would just need to make the function suspend and remove the `runBlocking` no?